api/src/components/flag/controller.ts

```diff
@@ -153,8 +163,18 @@ export const createFlagComment = async (
             });
         }
 
-        // The user attempting to leave a comment, is not the flag creator, or the publication owner
-        if (flag.createdBy !== event.user.id && flag.publication.user.id !== event.user.id) {
+        const publication = flag.publication;
+
+        // The user attempting to leave a comment is not the flag creator or the publication owner.
+        const latestPublishedVersion =
+            publication.versions.find((version) => version.isCurrent && version.currentStatus === 'LIVE') ||
+            publication.versions.find((version) => version.versionNumber === publication.versions.length - 1);
```

Comment:
the same as above, we can only allow users to flag the latest published version

Reply:
This has been addressed with the new isLatestLiveVersion field now.